config/crd/bases/postgres-operator.crunchydata.com_postgresclusters.yaml

```
@@ -110,11 +110,25 @@ spec:
                       x-kubernetes-map-type: atomic
                       x-kubernetes-validations:
                       - message: '"hba" cannot be combined with other fields'
-                        rule: 'has(self.hba) ? !has(self.connection) && !has(self.databases)
-                          && !has(self.method) && !has(self.options) && !has(self.users)
-                          : true'
+                        rule: '[has(self.hba), has(self.connection) || has(self.databases)
+                          || has(self.method) || has(self.options) || has(self.users)].exists_one(b,b)'
                       - message: '"connection" and "method" are required'
-                        rule: 'has(self.hba) ? true : has(self.connection) && has(self.method)'
+                        rule: has(self.hba) || (has(self.connection) && has(self.method))
+                      - message: the "ldap" method requires an "ldapbasedn", "ldapprefix",
+                          or "ldapsuffix" option
+                        rule: has(self.hba) || self.method != "ldap" || (has(self.options)
```

Do I understand this to say that either we have hba or we have ldap with ldap_ options? Is this another way to say that? 

```suggestion
                        rule: has(self.hba) || self.method == "ldap" && (has(self.options)
```

The "A or not B or C (required by B)" feels a little harder to read for me, but am I missing something?

---

Every rule must evaluate to true. `self.method == "ldap"` rejects all other methods.

---

I read these as "valid when (1) has hba or (2) isn't ldap or (3) [ldap stuff]"

---

> we have hba or we have ldap with ldap options

Nope. This is saying:

when [we don't have hba] and [the method is ldap],
then [options must be] and [options must blah blah];
otherwise [ok]

But that "otherwise" is _not_ "otherwise [the whole thing]." So, it's spelled:

when [we have hba] or [the method isn't ldap], [ok];
otherwise, [options must be] and [options must blah blah].